rbac/api/cross_access/serializer.py

```diff
@@ -106,27 +108,59 @@ def get_roles(self, obj):
         serialized_roles = [RoleSerializer(role).data for role in obj.roles.all()]
         return serialized_roles
 
+    def throw_validation_error(self, source, message):
+        """Construct a validation error and raise the error."""
+        error = {source: [message]}
+        raise ValidationError(error)
+
+    def validate_roles(self, roles):
+        """Format role list as expected for cross-account-request."""
+        public_tenant = Tenant.objects.get(tenant_name="public")
+
+        role_display_names = [role["display_name"] for role in roles]
```

Comment:
Could we put field `system` in the RoleSerializer so the roles will contain system info, if any of them is False, raise error?
If we can, then we don't need to run extra code below and raise error earlier

Reply:
I tried but it looks it will require to have system attribute as input or we would need to fetch role in `to_internal_value`  to get system variable or am I   missing here  something?

Reply:
Oh, right, NVM. 👍